src/config.ts

```diff
@@ -0,0 +1,86 @@
+import {AuthenticationConfig, ServiceConfig} from '../third_party/types/common-types';
+
+// ProfileAgentConfig is the configuration for the profiler.
+export interface ProfileAgentConfig extends AuthenticationConfig {
+  // Log levels: 0-disabled,1-error,2-warn,3-info,4-debug.
+  logLevel?: number;
+
+  // Specifies the service contect with which profiles from this application
+  // will be associated.
+  serviceContext?: {
+    // The service name.
+    service?: string;
+
+    // The service version.
+    version?: string;
+
+    // Unique deployment identifier.
+    minorVersion?: string;
+  };
+
+  // Configuration for heap profiling.
+  heapProfilerConfig?: HeapProfilerConfig;
```

Comment:
`Config` suffix in the property name seems a redundant. I would expect that in most usage this property would look like `config.heapProfilerConfig`. It could be argued that even the 'Profiler` suffix is a bit too verbose, but I'll leave that up to you.